src/dune_rules/pkg_rules.ml

```diff
@@ -651,12 +655,17 @@ module Action_expander = struct
           ])
   ;;
 
+  (* An action which has no effect. Note that we can't use [Action.Progn []]
+     here because currently every rule must have at least one action that
+     requires sandboxing. *)
+  let action_noop = Action.System "true"
```

Comment:
Instead of adding the hack in the code, you could just add it in the test. E.g.

```
(progn
  (system "true")
  (when ...))
```

All real builds are going to have at least 1 run action, so this limitation isn't going to matter much.

Reply:
That wouldn't test anything not already tested in the previous test so I'll just remove the empty case then.